# DATA ANALYTICS LOGISTICS

In [1]:
import plotly.express as px
import plotly.graph_objects as go
import pandas as pd

### Importando a base

In [22]:
df = pd.read_csv("Base/bd.csv")

### Composição de coluna e head

In [23]:
df.head()

ID_transporte                 data  tempo_viagem  velocidade_media  \
0              1  2022-01-01 08:15:00            30                60   
1              2  2022-01-02 14:30:00            45                55   
2              3  2022-01-02 10:45:00            20                40   
3              4  2022-01-03 09:20:00            35                50   
4              5  2022-01-04 12:10:00            50                45   

   consumo_combustivel  carga_transportada  ID_motorista  feedback_cliente  
0                   10                 100           101               4.5  
1                   15                 200           102               4.2  
2                    8                  50           103               4.8  
3                   12                 150           101               3.9  
4                   18                 250           104               4.6

### Tipos de dados

In [24]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100 entries, 0 to 99
Data columns (total 8 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   ID_transporte        100 non-null    int64  
 1   data                 100 non-null    object 
 2   tempo_viagem         100 non-null    int64  
 3   velocidade_media     100 non-null    int64  
 4   consumo_combustivel  100 non-null    int64  
 5   carga_transportada   100 non-null    int64  
 6   ID_motorista         100 non-null    int64  
 7   feedback_cliente     100 non-null    float64
dtypes: float64(1), int64(6), object(1)
memory usage: 6.4+ KB


### Principais informações sobre a base

In [25]:
df.describe()

ID_transporte  tempo_viagem  velocidade_media  consumo_combustivel  \
count     100.000000    100.000000         100.00000           100.000000   
mean       50.500000     39.750000          46.85000            13.490000   
std        29.011492      9.544553           7.83849             3.355969   
min         1.000000     20.000000          35.00000             8.000000   
25%        25.750000     30.000000          40.00000            10.000000   
50%        50.500000     40.000000          45.00000            13.000000   
75%        75.250000     46.250000          51.25000            15.500000   
max       100.000000     55.000000          60.00000            20.000000   

       carga_transportada  ID_motorista  feedback_cliente  
count          100.000000    100.000000        100.000000  
mean           165.800000    104.250000          4.458000  
std             64.810212      2.266912          0.280036  
min             50.000000    101.000000          3.700000  
25%            100.000000    102.000000          4.200000  
50%            150.000000    104.000000          4.500000  
75%            205.000000    106.000000          4.700000  
max            300.000000    108.000000          4.900000

### Verificando a presenção de dados faltantes

In [26]:
df.isna().sum()

ID_transporte          0
data                   0
tempo_viagem           0
velocidade_media       0
consumo_combustivel    0
carga_transportada     0
ID_motorista           0
feedback_cliente       0
dtype: int64

### Observar alguns comportamentos iniciais dos dados

#### Existem 11 tipos de cargas a serem transportadas

##### Podemos observar algumas tendências para cada tipo de carga

In [27]:
df['carga_transportada'].unique()

array([100, 200,  50, 150, 250,  80, 180, 300, 120, 220, 280], dtype=int64)

In [59]:
def grafico_hist(x):
  fig=px.histogram(df, x=x)
  fig.show()

x='carga_transportada'
grafico_hist(x)

### Apresentar os detalhes principais e iremos depois correlacionar com o tipo de carga, entender se muda algo ou não

In [54]:
def DescricaoBasica(Column):
    media = df[Column].mean()  
    mediana = df[Column].median()  
    desvio_padrao = df[Column].std() 
    min = df[Column].min() 
    max = df[Column].max() 

    return {
        'Média': media,
        'Mediana': mediana,
        'Desvio Padrão': desvio_padrao,
        'Mínimo': min,
        'Máximo': max
    }

In [55]:
DescricaoBasica('consumo_combustivel')

{'Média': 13.49,
 'Mediana': 13.0,
 'Desvio Padrão': 3.355968601242457,
 'Mínimo': 8,
 'Máximo': 20}

In [56]:
DescricaoBasica('feedback_cliente')

{'Média': 4.458,
 'Mediana': 4.5,
 'Desvio Padrão': 0.28003607271243114,
 'Mínimo': 3.7,
 'Máximo': 4.9}

In [57]:
DescricaoBasica('tempo_viagem')

{'Média': 39.75,
 'Mediana': 40.0,
 'Desvio Padrão': 9.544552626943016,
 'Mínimo': 20,
 'Máximo': 55}

In [58]:
DescricaoBasica('velocidade_media')

{'Média': 46.85,
 'Mediana': 45.0,
 'Desvio Padrão': 7.838489598890795,
 'Mínimo': 35,
 'Máximo': 60}

In [61]:
resultados = df.groupby('carga_transportada').agg(
    media_velocidade=('velocidade_media', 'mean'),
    media_consumo=('consumo_combustivel', 'mean'),
    media_tempo_viagem=('tempo_viagem', 'mean'),
    feedback_medio=('feedback_cliente', 'mean'),
    count=('ID_transporte', 'count')  
).reset_index()

resultados

carga_transportada  media_velocidade  media_consumo  media_tempo_viagem  \
0                   50         40.000000       8.000000           20.000000   
1                   80         59.583333       8.916667           25.000000   
2                  100         54.615385      10.000000           30.000000   
3                  120         50.416667      11.166667           35.000000   
4                  150         45.384615      12.923077           39.615385   
5                  180         40.000000      14.000000           40.000000   
6                  200         50.000000      15.000000           45.000000   
7                  220         40.000000      17.090909           50.000000   
8                  250         42.500000      18.000000           50.000000   
9                  280         35.000000      19.000000           55.000000   
10                 300         35.000000      20.000000           55.000000   

    feedback_medio  count  
0         4.800000      1  
1         4.516667     12  
2         4.507692     13  
3         4.466667     12  
4         4.400000     13  
5         4.400000     12  
6         4.425000     12  
7         4.490909     11  
8         4.250000      2  
9         4.460000     10  
10        4.500000      2

## **Carga 50**:
### **Observação**: A velocidade média de 40 km/h é muito baixa para um tempo de viagem de 20 dias, sugerindo que a entrega foi de fato uma viagem muito longa ou com muitas paradas.
### **Feedback Alto**: Mesmo com um tempo tão longo, o feedback médio de 4.8 sugere que a entrega foi considerada eficiente ou satisfatória, possivelmente devido à sua previsibilidade ou condições favoráveis.
### **Conclusão**: O alto feedback pode indicar que os clientes valorizam a eficiência da entrega, mesmo em longos períodos, contanto que os resultados finais atendam às expectativas.

## **Tendência Geral**:
### **Carga Pesada (250 e 300)**: Para cargas mais pesadas, com um tempo de viagem de 55 dias, a velocidade média se mantém em 35 km/h, e o feedback médio é mais baixo (4.25 - 4.50).
### **Análise**: Essa relação sugere que à medida que o tempo de entrega aumenta, a satisfação do cliente diminui, mesmo que a entrega seja feita com uma velocidade relativamente constante.

## **Relações Entre Variáveis**:

### **Velocidade e Consumo**: A velocidade média permanece baixa à medida que o tempo de viagem se estende. O tempo de viagem prolongado pode significar mais paradas, aumento do consumo de combustível e custos elevados.
### **Feedback e Tempo**: Um feedback médio decrescente com o aumento do tempo de viagem sugere que entregas mais longas impactam negativamente a percepção do cliente, possivelmente pela espera e pelo custo.

In [62]:
def grafico_bar(eixoX, eixoY): 
    fig = px.bar(df, x=eixoX, y=eixoY)
    fig.update_traces(marker_color='#1db954')
    fig.update_layout(
        title_text='Velocidade Média por Carga Transportada',
        title_font_family='Courier New',
        title_font_color='#1db954',
        plot_bgcolor='#191414',
        height=600,
        width=1000
    )
    fig.update_xaxes(showgrid=False, zeroline=False, visible=True)
    fig.update_yaxes(showgrid=False, zeroline=False, visible=False, hoverformat=False)
    fig.show()

x = 'carga_transportada'
y = 'velocidade_media'

grafico_bar(x, y)

In [63]:
def grafico_dispercao(eixoX, eixoY, eixoZ):
    fig = px.scatter(df, x=eixoX, y=eixoY, size=eixoZ, 
                     title='Relação entre Tempo de Viagem, Consumo de Combustível e Carga Transportada',
                     labels={eixoX: 'Tempo de Viagem (dias)', eixoY: 'Consumo de Combustível (litros)', eixoZ: 'Carga Transportada'},
                     hover_name='ID_transporte')
    fig.update_layout(title_font_family='Courier New',
                      title_font_color='#1db954',
                      plot_bgcolor='#191414',
                      height=600,
                      width=1000)
    fig.show()

x = 'tempo_viagem'
y = 'consumo_combustivel'
z = 'carga_transportada'

grafico_dispercao(x, y, z)

In [64]:
def grafico_pizza(eixoX, eixoY):
    df_emp = (df
              .groupby([eixoX])[eixoY]
              .mean()  # Calcula a média do feedback por carga
              .to_frame()
              .reset_index()
             )

    fig = px.pie(df_emp, values=eixoY, names=eixoX,
                 title='Distribuição do Feedback Médio por Carga Transportada')
    fig.update_traces(marker=dict(line=dict(color='#191414', width=2)))  # Adiciona contorno
    fig.show()

x = 'carga_transportada'
y = 'feedback_cliente'

grafico_pizza(x, y)
